x/distribution/client/cli/tx.go

```diff
@@ -150,3 +152,65 @@ $ %s tx set-withdraw-addr cosmos1gghjut3ccd8ay0zduzj64hwre2fxs9ld75ru9p --from m
 	}
 	return cmd
 }
+
+// GetCmdSubmitProposal implements the command to submit a community-pool-spend proposal
+func GetCmdSubmitProposal(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "community-pool-spend [proposal-file]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Submit a community pool spend proposal",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Submit a community pool spend proposal along with an initial deposit.
+The proposal details must be supplied via a JSON file.
+
+Example:
+$ %s tx gov submit-proposal community-pool-spend <path/to/proposal.json> --from=<key_or_address>
+
+Where proposal.json contains:
+
+{
+  "title": "Community Pool Spend",
+  "description": "Pay me some Atoms!",
+  "recipient": "cosmos1s5afhd6gxevu37mkqcvvsj8qeylhn0rz46zdlq",
+  "amount": [
+    {
+      "denom": "stake",
+      "amount": "10000"
+    }
+  ],
+  "deposit": [
+    {
+      "denom": "stake",
+      "amount": "10000"
+    }
+  ]
+}
+`,
+				version.ClientName,
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			txBldr := authtxb.NewTxBuilderFromCLI().WithTxEncoder(utils.GetTxEncoder(cdc))
+			cliCtx := context.NewCLIContext().
+				WithCodec(cdc).
+				WithAccountDecoder(cdc)
+
+			proposal, err := distrcutils.ParseCommunityPoolSpendProposalJSON(cdc, args[0])
+			if err != nil {
+				return err
+			}
+
+			from := cliCtx.GetFromAddress()
+			content := types.NewCommunityPoolSpendProposal(proposal.Title, proposal.Description, proposal.Recipient, proposal.Amount)
```

Comment:
It would be nice for `NewCommunityPoolSpendProposal` to just (only) take the `proposal` object here.

Reply:
I don't quite understand? This is a `New` function which returns the proposal object, there is no proposal object yet to pass.

Reply:
you have defined an object named `proposal` which already being referenced here (eg. `proposal.Title`) that's what I mean by proposal

Reply:
ah, yes, but that is a slightly different struct parsed from the provided JSON file

Reply:
Yeah I think this is fine as-is. The `proposal` object here comes from `distrcutils` and we're passing those fields to a function in `types`. Would create an awkward package dependency.

Reply:
good point on the dependancy